src/control-plane/backend/lambda/api/test/api/reporting.test.ts

```diff
@@ -614,6 +614,106 @@ describe('reporting test', () => {
 
   });
 
+  it('retention visual - publish', async () => {
+    tokenMock(ddbMock, false);
+    stsClientMock.on(AssumeRoleCommand).resolves({
+      Credentials: {
+        AccessKeyId: '1111',
+        SecretAccessKey: '22222',
+        SessionToken: '33333',
+        Expiration: new Date(),
+      },
+    });
+
+    redshiftClientMock.on(BatchExecuteStatementCommand).resolves({
+    });
+    redshiftClientMock.on(DescribeStatementCommand).resolves({
+      Status: StatusString.FINISHED,
+    });
+
+    quickSightMock.on(CreateAnalysisCommand).resolves({
+      Arn: 'arn:aws:quicksight:us-east-1:11111111:analysis/analysisaaaaaaaa',
+    });
+    quickSightMock.on(CreateDashboardCommand).resolves({
+      Arn: 'arn:aws:quicksight:us-east-1:11111111:dashboard/dashboard-aaaaaaaa',
+      VersionArn: 'arn:aws:quicksight:us-east-1:11111111:dashboard/dashboard-aaaaaaaa/1',
+    });
+
+    const res = await request(app)
+      .post('/api/reporting/retention')
+      .send({
+        action: 'PUBLISH',
+        viewName: 'testview0002',
+        projectId: 'project01_wvzh',
+        pipelineId: 'pipeline-1111111',
+        appId: 'app1',
+        sheetName: 'sheet99',
+        computeMethod: 'USER_CNT',
+        specifyJoinColumn: true,
+        joinColumn: 'user_pseudo_id',
+        conversionIntervalType: 'CUSTOMIZE',
+        conversionIntervalInSeconds: 7200,
+        eventAndConditions: [{
+          eventName: 'add_button_click',
```

Comment:
add test case with some conditions

Reply:
The conditions have already been covered in other test cases, adding more conditions in this test case won't have any effect. It only affects the SQL statement, which is already covered in sql-builder.test.ts.